standards/revolve2/standards/genotypes/cppnwin/modular_robot/v2/_body_develop.py

```diff
@@ -128,33 +137,27 @@ def __add_child(
     if grid[tuple(position)] > 0:
         return None
 
-    """Now we anjust the position for the potential new module to fit the attachment point of the parent, additionally we query the CPPN for child type and angle of the child."""
+    """Now we adjust the position for the potential new module to fit the attachment point of the parent."""
     new_pos = np.array(np.round(position + attachment_point.offset), dtype=np.int64)
     child_type, angle = __evaluate_cppn(body_net, new_pos, chain_length)
 
-    """Here we check whether the CPPN evaluated to place a module and if the module can be set on the parent."""
+    """Check if we can place a child and if so, create the module."""
     can_set = module.module_reference.can_set_child(attachment_index)
     if (child_type is None) or (not can_set):
-        return None  # No module will be placed.
+        return None
 
     """Now we know we want a child on the parent and we instantiate it, add the position to the grid and adjust the up direction for the new module."""
     child = child_type(angle)
     grid[tuple(position)] += 1
     up = __rotate(module.up, forward, Quaternion.from_eulers([angle, 0, 0]))
     module.module_reference.set_child(child, attachment_index)
 
-    return __Module(
-        position,
-        forward,
-        up,
-        chain_length,
-        child,
-    )
+    return __Module(position, forward, up, chain_length, child)
 
 
 def __rotate(a: Vector3, b: Vector3, rotation: Quaternion) -> Vector3:
     """
-    Rotates vector a, a given angle around b.
+    Rotate vector a, a given angle around b.
```

Comment:
Hard agree, I feel like by both rounding and ALSO casting to int we are approximating the rotation to a discrete step, rather than smoothly rotating which is the whole benefit of Rodrigues and quaternions. I will remove the rounding and use float instead.

Reply:
Actually, I think this is to ensure that the angle remains one of (0, 90, 180, 270). Since this rotate is only used for the placement of modules, it seems the "loss of precision" is intended to keep the angle a discrete multiple of 90.